### PR TITLE
[FIX] Scatter Plot: VizRank disabled when no class vars

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
**Scatter Plot** VizRank crashes when _class column_ has only `nan` values.

##### Description of changes
VizRank is disabled as it is when there is no `class_var`.

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/401246621/

File: Orange/widgets/visualize/tests/test_owscatterplot.py [hunk 1]
```diff
@@ -392,6 +392,27 @@ def test_vizrank(self):
...
+            self.send_signal(self.widget.Inputs.data, data)
+            self.assertEqual(is_enabled, self.widget.vizrank_button.isEnabled())
+
+        data1 = Table("iris")[::30]
+        data2 = Table("iris")[::30]
+        data2.Y[:] = np.nan
+        domain = Domain(
+            attributes=data2.domain.attributes[:4], class_vars=DiscreteVariable("iris", values=[]))
+        data2 = Table(domain, data2.X, Y=data2.Y)
+        data3 = Table("iris")[::30]
```
Comment (lanzagar) [inline Orange/widgets/visualize/tests/test_owscatterplot.py]: What is the difference between data1 and data3?  (i.e. why this line)

File: Orange/widgets/visualize/owscatterplot.py [hunk 2]
```diff
@@ -252,6 +252,8 @@ def set_data(self, data):
         self.vizrank_button.setEnabled(
             self.data is not None and not self.data.is_sparse() and
             self.data.domain.class_var is not None and
+            (len(self.data.domain.class_var.values)
+             if self.data.domain.class_var.is_discrete else True) and
```
Comment (lanzagar) [inline Orange/widgets/visualize/owscatterplot.py]: This fixes the case when the class has no values. How about when the class has e.g. 3 values, but all instances have unknown classes (e.g. a test set with the same variable as the training set, but class unknown)

File: Orange/widgets/visualize/tests/test_owscatterplot.py (hunk 1 above)
Comment (jerneju) [reply Orange/widgets/visualize/tests/test_owscatterplot.py]: Now it is the difference.
